options.go

```diff
@@ -99,6 +100,13 @@ func WithUpstreamOptions(options ...temporal.ServerOption) ServerOption {
 	})
 }
 
+// WithBaseConfig imports configuration from a standard temporal configuration file.
```

Comment:
Doc nit: this option accepts a Go type, not a file path.

```suggestion
// WithBaseConfig sets the default Temporal server configuration.
//
// Storage and client configuration will always be overridden, however base config can be
// used to enable settings like TLS or authentication.
```